core/src/main/java/hudson/PluginStrategy.java

```diff
@@ -92,5 +92,5 @@ PluginWrapper createPluginWrapper(File archive)
      * @param dependee newly loaded plugin.
      * @since 1.557
      */
-    void updateDependency(PluginWrapper depender, PluginWrapper dependee);
+    default void updateDependency(PluginWrapper depender, PluginWrapper dependee) {}
```

Comment:
Annotation is not a blocker for this change, but it definitely makes sense to create a follow-up JIRA ticket. We have a bunch of use-cases in the core code.